### ISSUE TITLE
Can you just ignore a file rather than the entire module?

### ISSUE THREAD

**Issue by sunqingquan** [rockbruno/swiftshield#45]
Can you just ignore a file rather than the entire module?
e.g. I have a module called "AA "which contains 10 files, such as a.swift, b.swift ... , I just want to ignore the file a.swift, and the other files need to be confused.

Regard

**Comment by rockbruno** [comment]
You can't do it right now, but it sounds like something that wouldn't be difficult do develop :)

**Comment by sunqingquan** [comment]
Hopefully you can do that, or I'll need to write additional scripts to accommodate it.

Regards

**Comment by pajtimid** [comment]
Any option for excluding swift files from obfuscation

**Comment by rockbruno** [comment]
Can you guys give me an example on why you want to do this? Many people require this but I don't understand the purpose, unless the file is completely independent from the rest of the project, this will result in the project not building. Do you mean ignoring symbols *created* in a specific file?

**Comment by pajtimid** [comment]
I need this, because I want to ignore Core Data swift classes, because in data model doesn't obfuscate Entities. Also Core Store pod have so many issues during obfuscating. So I need to exclude Core Data classes.

**Comment by rockbruno** [comment]
Makes sense, I didn't consider CoreData. I'll see if it's possible to obfuscate the entities, that would solve the issue as well. By the way, can you add core data models to other modules? If so, you can circumvent this by simply adding the files you don't want to obfuscate to a new module and ignore the entire module.

**Comment by pajtimid** [comment]
Ok, I will check if I can add core data models to other modules. 
Thank you for your support.

**Comment by pajtimid** [comment]
Any news about obfuscating the entities on Core Data ?

**Comment by rockbruno** [comment]
Although obfuscating the models itself would be simple, I'm not sure if this would work as CoreData uses the name of the objects as keys... I believe that if you push a new version with a different obfuscation, the database will stop working. 

If that's true then the best approach imo is to ignore these files either by pushing them to another module, or by finally developing this feature :)

I won't work on this, but if someone wants to take shot, I think all you need to do is make an argument that gives you a `Set` of the files you want to ignore (see `main.swift` for the other arguments) and then add a condition to not recurse the file when searching for declarations here: https://github.com/rockbruno/swiftshield/blob/master/swiftshield-Sources/AutomaticSwiftShield.swift#L70

**Comment by iOSleep** [comment]
@pajtimid a simple way, just cover the obfuscationDict 

https://github.com/rockbruno/swiftshield/blob/master/swiftshield-Sources/AutomaticSwiftShield.swift#L76

```swift
let name = data.name
// coredata genrate file use origin name
if file.path.contains("DerivedSources/CoreDataGenerated") {
    obfuscationData.obfuscationDict[name] = name
}
let usr = data.usr
```
